contracts/fio.address/fio.address.cpp

```diff
@@ -669,6 +727,134 @@ namespace fioio {
+
+            auto domainsbyname = domains.get_index<"byname"_n>();
+            auto domains_iter = domainsbyname.find(domainHash);
+
+            fio_400_assert(domains_iter != domainsbyname.end(), "fio_address", fa.fioaddress,
+                           "FIO Domain not registered",
+                           ErrorDomainNotRegistered);
+
+            //add 30 days to the domain expiration, this call will work until 30 days past expire.
+            const uint32_t domain_expiration = get_time_plus_seconds(domains_iter->expiration, SECONDS30DAYS);
+
+            const uint32_t present_time = now();
+            fio_400_assert(present_time <= domain_expiration, "fio_address", fa.fioaddress, "FIO Domain expired",
+                           ErrorDomainExpired);
+
+            auto namesbyname = fionames.get_index<"byname"_n>();
+            auto fioname_iter = namesbyname.find(nameHash);
+            fio_400_assert(fioname_iter != namesbyname.end(), "fio_address", fa.fioaddress,
+                           "FIO address not registered", ErrorFioNameNotRegistered);
+            fio_403_assert(fioname_iter->owner_account == actor.value,
+                           ErrorSignature); // check if actor owns FIO Address
+
+
+
+
+            //FEE PROCESSING
+            uint64_t fee_amount = 0;
+
+            //begin fees, bundle eligible fee logic
+            const uint128_t endpoint_hash = string_to_uint128_hash(UPDATE_ENCRYPT_KEY_ENDPOINT);
+
+            auto fees_by_endpoint = fiofees.get_index<"byendpoint"_n>();
+            auto fee_iter = fees_by_endpoint.find(endpoint_hash);
+
+            //if the fee isnt found for the endpoint, then 400 error.
+            fio_400_assert(fee_iter != fees_by_endpoint.end(), "endpoint_name", REMOVE_PUB_ADDRESS_ENDPOINT,
+                           "FIO fee not found for endpoint", ErrorNoEndpoint);
+
+            const int64_t reg_amount = fee_iter->suf_amount;
+            const uint64_t fee_type = fee_iter->type;
+
+            fio_400_assert(fee_type == 1, "fee_type", to_string(fee_type),
+                           "update_encrypt_key unexpected fee type for endpoint update_encrypt_key, expected 1",
+                           ErrorNoEndpoint);
+
+            const uint64_t bundleeligiblecountdown = fioname_iter->bundleeligiblecountdown;
+
+            if (bundleeligiblecountdown > 0) {
+                namesbyname.modify(fioname_iter, _self, [&](struct fioname &a) {
+                    a.bundleeligiblecountdown = (bundleeligiblecountdown - 1);
+                });
+            } else {
+                fee_amount = fee_iter->suf_amount;
+                fio_400_assert(max_fee >= (int64_t) fee_amount, "max_fee", to_string(max_fee),
+                               "Fee exceeds supplied maximum.",
+                               ErrorMaxFeeExceeded);
+
+                //NOTE -- question here, should we always record the transfer for the fees, even when its zero,
```

Comment:
Adding comment to make sure this gets answered by @pawmmm

Reply:
is there any action for this?